diff --git a/android-configure b/android-configure
index 43341d1abe..11a916cb76 100755
--- a/android-configure
+++ b/android-configure
@@ -26,7 +26,7 @@ CC_VER="4.9"
 case $ARCH in
     arm)
         DEST_CPU="arm"
-        TOOLCHAIN_NAME="armv7-linux-androideabi"
+        TOOLCHAIN_NAME="armv7a-linux-androideabi"
         ;;
     x86)
         DEST_CPU="ia32"
@@ -56,10 +56,10 @@ export CXX_host=$(command -v g++)
 host_gcc_version=$($CC_host --version | grep gcc | awk '{print $NF}')
 major=$(echo $host_gcc_version | awk -F . '{print $1}')
 minor=$(echo $host_gcc_version | awk -F . '{print $2}')
-if [ -z $major ] || [ -z $minor ] || [ $major -lt 6 ] || ( [ $major -eq 6 ] && [ $minor -lt 3 ] ); then
-  echo "host gcc $host_gcc_version is too old, need gcc 6.3.0"
-  return 1
-fi
+#if [ -z $major ] || [ -z $minor ] || [ $major -lt 6 ] || ( [ $major -eq 6 ] && [ $minor -lt 3 ] ); then
+#  echo "host gcc $host_gcc_version is too old, need gcc 6.3.0"
+#  return 1
+#fi
 
 SUFFIX="$TOOLCHAIN_NAME$ANDROID_SDK_VERSION"
 TOOLCHAIN=$NDK_PATH/toolchains/llvm/prebuilt/$HOST_OS-$HOST_ARCH
@@ -70,7 +70,7 @@ export CXX=$TOOLCHAIN/bin/$SUFFIX-clang++
 
 
 GYP_DEFINES="target_arch=$ARCH"
-GYP_DEFINES+=" v8_target_arch=$ARCH"
+GYP_DEFINES+=" v8_target_arch=$DEST_CPU"
 GYP_DEFINES+=" android_target_arch=$ARCH"
 GYP_DEFINES+=" host_os=$HOST_OS OS=android"
 export GYP_DEFINES
