Index: vscode/node-src/deps/v8/src/logging/log.cc
===================================================================
--- vscode.orig/node-src/deps/v8/src/logging/log.cc
+++ vscode/node-src/deps/v8/src/logging/log.cc
@@ -308,7 +308,7 @@ class PerfBasicLogger : public CodeEvent
   FILE* perf_output_handle_;
 };
 
-const char PerfBasicLogger::kFilenameFormatString[] = "/tmp/perf-%d.map";
+const char PerfBasicLogger::kFilenameFormatString[] = "/data/data/vn.vhn.vsc/files/tmp/perf-%d.map";
 // Extra space for the PID in the filename
 const int PerfBasicLogger::kFilenameBufferPadding = 16;
 
