Index: vscode/node-src/deps/v8/src/logging/log.cc
===================================================================
--- vscode.orig/node-src/deps/v8/src/logging/log.cc
+++ vscode/node-src/deps/v8/src/logging/log.cc
@@ -367,7 +367,7 @@ class LinuxPerfBasicLogger : public Code
   static uint64_t reference_count_;
 };
 
-const char LinuxPerfBasicLogger::kFilenameFormatString[] = "/tmp/perf-%d.map";
+const char LinuxPerfBasicLogger::kFilenameFormatString[] = "/data/data/vn.vhn.vsc/files/tmp/perf-%d.map";
 // Extra space for the PID in the filename
 const int LinuxPerfBasicLogger::kFilenameBufferPadding = 16;
 
