Index: vscode/node-src/deps/npm/lib/cli/entry.js
===================================================================
--- vscode.orig/node-src/deps/npm/lib/cli/entry.js
+++ vscode/node-src/deps/npm/lib/cli/entry.js
@@ -67,9 +67,9 @@ module.exports = async (process, validat
     // this is async but we dont await it, since its ok if it doesnt
     // finish before the command finishes running. it uses command and argv
     // so it must be initiated here, after the command name is set
-    const updateNotifier = require('./update-notifier.js')
+    // const updateNotifier = require('./update-notifier.js')
     // eslint-disable-next-line promise/catch-or-return
-    updateNotifier(npm).then((msg) => (npm.updateNotification = msg))
+    // updateNotifier(npm).then((msg) => (npm.updateNotification = msg))
 
     await execPromise
     return exitHandler.exit()
