Index: vscode/node-src/lib/child_process.js
===================================================================
--- vscode.orig/node-src/lib/child_process.js
+++ vscode/node-src/lib/child_process.js
@@ -636,10 +636,8 @@ function normalizeSpawnArguments(file, a
     } else {
       if (typeof options.shell === 'string')
         file = options.shell;
-      else if (process.platform === 'android')
-        file = '/system/bin/sh';
       else
-        file = '/bin/sh';
+        file = '/data/data/vn.vhn.vsc/files/usr/bin/sh';
       args = ['-c', command];
     }
   }
