Index: vscode/node-src/lib/os.js
===================================================================
--- vscode.orig/node-src/lib/os.js
+++ vscode/node-src/lib/os.js
@@ -191,7 +191,7 @@ function tmpdir() {
     path = safeGetenv('TMPDIR') ||
            safeGetenv('TMP') ||
            safeGetenv('TEMP') ||
-           '/tmp';
+           '/data/data/vn.vhn.vsc/files/tmp';
     if (path.length > 1 && StringPrototypeEndsWith(path, '/'))
       path = StringPrototypeSlice(path, 0, -1);
   }
