Index: vscode/node-src/tools/v8_gypfiles/toolchain.gypi
===================================================================
--- vscode.orig/node-src/tools/v8_gypfiles/toolchain.gypi
+++ vscode/node-src/tools/v8_gypfiles/toolchain.gypi
@@ -62,7 +62,7 @@
     'mips_use_msa%': 0,
 
     # Print to stdout on Android.
-    'v8_android_log_stdout%': 0,
+    'v8_android_log_stdout%': 1,
 
     'v8_enable_backtrace%': 0,
 
